clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -6155,6 +6155,44 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
     return RValue::get(Builder.CreateIntrinsic(
         llvm::Intrinsic::cheri_cap_perms_check, {SizeTy}, {Cap, Perms}));
   }
+  case Builtin::BI__builtin_cheriot_sealing_type: {
+    const auto *ArgLit = dyn_cast<StringLiteral>(E->getArg(0));
+    assert(ArgLit && "Argument to built-in should be a string literal!");
+    auto SealingTypeName = ArgLit->getString().str();
+    SealingTypeName.erase(
```

Comment:
Should it just be an error for it to contain spaces? Maybe we should restrict it to be alphanumeric only?

Reply:
(removed)